src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: Array<T>, key: keyof T): GroupsMap<T> {
+  const groupedItems: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const itemKey = String(item[key]);
+
+    if (!groupedItems[itemKey]) {
+      groupedItems[itemKey] = [item];
+    } else {
```

Comment:
You could not use this else. Just write it down below without else.

Reply:
yep, you don't need else here